### PR TITLE
Fixes for oh-stepper & oh-input

### PR DESCRIPTION
**Pull request by ghys**
Improve date handling in inputs.

Signed-off-by: Yannick Schaus <github@schaus.net>

### CODE REVIEW

File: bundles/org.openhab.ui/web/src/components/widgets/system/oh-input.vue [hunk 1]
```diff
@@ -21,37 +17,65 @@
...
 
 export default {
   mixins: [mixin],
   widget: OhInputDefinition,
-  mounted () {
-    delete this.config.value
+  data () {
+    return {
+      pendingUpdate: null
+    }
```
Comment (mherwege) [inline bundles/org.openhab.ui/web/src/components/widgets/system/oh-input.vue:31]: Thanks forr this. I was preparing a PR as the context value should indeed not be used.

File: bundles/org.openhab.ui/web/src/components/widgets/system/oh-input.vue [hunk 2]
```diff
@@ -21,37 +17,65 @@
...
   computed: {
     value () {
       if (this.config.variable && this.context.vars[this.config.variable] !== undefined) {
         return this.context.vars[this.config.variable]
-      } else if (this.config.showOK && this.context.vars[this.config.value] !== undefined) {
-        return this.context.vars[this.config.value]
+      } else if (this.config.sendButton && this.pendingUpdate !== null) {
+        return this.pendingUpdate
       }
       return (this.config.item) ? this.context.store[this.config.item].state : this.config.defaultValue
```
Comment (mherwege) [inline bundles/org.openhab.ui/web/src/components/widgets/system/oh-input.vue:40]: I had it changed to something like this (which fit my use case):
```
      } else if (this.config.item && (this.context.store[this.config.item].state !== "NULL") && (this.context.store[this.config.item].state !== "UNDEF")) {
        return this.context.store[this.config.item].state
      }
      return this.config.defaultValue
```

File: bundles/org.openhab.ui/web/src/components/widgets/system/oh-input.vue [hunk 3]
```diff
@@ -21,37 +17,65 @@
...
-    clicked () {
-      if (this.config.item) {
-        this.$store.dispatch('sendCommand', { itemName: this.config.item, cmd: this.context.vars[this.config.value] })
+    sendButtonClicked () {
+      if (this.config.item && this.pendingUpdate) {
+        let cmd = this.pendingUpdate
+        if (this.config.type === 'datepicker' && Array.isArray(cmd)) {
+          cmd = dayjs(cmd[0]).format()
+        }
+        this.$store.dispatch('sendCommand', { itemName: this.config.item, cmd })
```
Comment (mherwege) [inline bundles/org.openhab.ui/web/src/components/widgets/system/oh-input.vue:77]:
```suggestion
        this.$store.dispatch('sendCommand', { itemName: this.config.item, cmd: cmd })
```

File: bundles/org.openhab.ui/web/src/components/widgets/system/oh-input.vue (hunk 2 above)
Comment (ghys) [reply bundles/org.openhab.ui/web/src/components/widgets/system/oh-input.vue]: Yes, makes sense, thanks!

File: bundles/org.openhab.ui/web/src/components/widgets/system/oh-input.vue (hunk 3 above)
Comment (ghys) [reply bundles/org.openhab.ui/web/src/components/widgets/system/oh-input.vue:79]: That's the shorthand notation, it's fine ;)

**Pull request merged**